### PR TITLE
Add guest_deregister in zCC for unmanaging VM

### PR DESCRIPTION
**Pull request by SeanHQF**
The VM still exists after deregistered, but its record is
removed from zCC. This is typically used by user to dismiss/offboard
a VM when he doesn't want to manage it through zCC.

nova virt zvm driver should call this API to unmanage a VM with userid,
e.g. in hypervisor.py and driver.py.

Signed-off-by: QingFeng Hao <haoqf@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by SeanHQF** [comment]
> please add unit test, rest API update

Added, please check, thanks

**Comment by jichenjc** [comment]
ok, looks good  now

@bjhuangr can you take another look? Thanks

File: doc/source/restapi.rst [hunk 1]
```diff
@@ -898,6 +898,30 @@ Register guest to be managed by z/VM Cloud Connector.
...
+**POST /guests/{userid}/action**
+
+Deregister guest to be managed by z/VM Cloud Connector.
+
+* Request:
+
+.. restapi_parameters:: parameters.yaml
+
+  - userid: guest_userid
+  - action: action_deregister_guest
```
Comment (jichenjc) [inline doc/source/restapi.rst]: where's the action_deregister_guest defined?

Comment (SeanHQF) [reply doc/source/restapi.rst]: Sorry I missed that and now added it in parameters.yaml, thx

**Comment by SeanHQF** [comment]
@jichenjc can you pls help to merge it ? thanks

**Pull request merged**